### PR TITLE
Add allocate-action.md doc

### PR DESCRIPTION
**Pull request by zbbkeepgoing**
Allocate action doc not exist in design doc, so add it.

### CODE REVIEW

**Comment by wangyang0616** [comment]
At present, e2e may fail to download the pytorch image, which leads to CI failure. Can it be triggered to run CI once in the background? 
cc @william-wang

**Comment by hwdef** [comment]
please retrigger the CI, I think the CI is OK now.
I think the allocate action is the most important action, and its documentation should be more comprehensive.

File: docs/design/allocate-action.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+1. NamespaceOrderFn(Plugin: DRF),
+2. QueueOrderFn(Plugin: Priority, DRF, Proportion),
+3. TaskOrderFn(Plugin: Priority),
+4. JobOrderFn(Plugin: Priority, DRF, Gang),
+5. AllocatableFn(Plugin: Proportion),
+6. PrePredicateFn(Plugin: Predicates),
+7. PredicateFn(Plugin: Predicates),
+8. BatchNodeOrderFn(Plugin: NodeOrder)
+
+### 1. NamespaceOrderFn:
```
Comment (lowang-bh) [inline docs/design/allocate-action.md:31]: This one has been removed.

**Pull request closed without merging**